### ISSUE TITLE
Unphysical implementation of E&B09 spin-cut

### ISSUE THREAD

**Issue by vetlewi** [oslocyclotronlab/ompy#69]
The implementation of the E&B09 spin cut (eq. 16) is not physical when the excitation energy is less than the half the deuteron pairing energy, as the spin-cut becomes zero. A more physical shape would be a constant at energies below a certain threshold.

In the old counting code this has been handled by
```
Eeff = Ex - 0.5Pa'
if Eeff < 0.0050:
    Eeff = 0.0050
```
which is seems to be somewhat arbitrary. However, since there is in general a low level density at those energies it might not matter much.

**Comment by fzeiser** [comment]
I agree, that this may lead to unexpected behavior. In principal, if I Interpret the article correctly, the equation is defined for `eff > 2 pa` only. Do you agree? So it should be noted in the docs if we define it. We could also have it as some (default?) constant...

**Comment by vetlewi** [comment]
I can't see there being an explicit low limit for where the formula is valid, but surely the only appropriate would be to use a constant under a certain energy. The question is where that limit should be. Depending on the nucleus being odd-odd, odd-A or even-even the Pa' could be anything from -6 to 6 MeV. Take e.g. 58Fe where Pa' = 2.874 MeV, with the requirement that Eeff > 2Pa', then the excitation energy of 58Fe has to be at least 7.19 MeV before the formula to be valid, which seems to be a little excessive. I would think the solution used in counting is reasonable, though it should be noted that the limit is arbitrary.
Best would be to evaluate the decay width to discrete states separately in the evaluation of <Gg0>.

**Comment by fzeiser** [comment]
I don’t have access to so many documents, bad internet connection. But I’ll check at home: what is it that hinders us to include the discrete states? I agree that this would be cleaner. But probably we anyhow need to include one modeling for energies below twice the pairing energy.

> I can't see there being an explicit low limit for where the formula is valid
Locking at the figures, it seems to me that they don’t plot any spin cutoff below 2xpa. To some extend it make sense not to, as we said here, too. For the discrete, I expect another spin distribution than the bell shape with a width given my the spin cut.

But still, plot of the cumulative nld at https://www-nds.iaea.org/cgi-bin/ripl_level_plot.pl?Z=26&A=58&EX=auto, I agree with you critique that 7 MeV is a quite high start.

**Comment by fzeiser** [comment]
I should refine my comment above. It seems like they draw some arbitrary cut as well. If they were to have a minimum excitation energy, their spin cuts would go to 0 as Eeff goes to 0. This is crazy.

**Comment by fzeiser** [comment]
@cecgresonant: Any opinions on this from you side? 

I could also imagine to use the *low energy σ* of EB for those cases, see Fig 2 and 3. So my idea would be:

- Create two new models `EB05 + low Sigma` and `EB09 + low sigma`
- In each of these, say that σ is calculated according to the "low sigma" fix, when 1) the effective energy `Eff` is below 0 or 2) "low sigma" is higher than the "normal" result.
- If the user "just" uses `EB05` or `EB09` it shall return `np.nan` for the values with `Eff < 0`

Graphically speaking, the proposed change would result in a combination of the red dashed lines and the spin cut formulas that are not defined at low energies.

**Comment by fzeiser** [comment]
I refer to EB09; in principal Fig 2 - 5

**Comment by cecgresonant** [comment]
This is an interesting problem. If I remember correctly (will check later today), there is indeed an unphysical behavior for one of the spincut models, but I thought it was the von Egidy & Bucurescu 2005/2006 one, and not the phenomenological formula from 2009? Maybe it is both of them... I will check ASAP :)

**Comment by fzeiser** [comment]
The issue is for both of them: What to do when `Eff` goes below 0.

**Comment by cecgresonant** [comment]
Here is what is written in E&B, PRC 72 (2005), Sec. II A 1.:
"The function exp(2√ax)x^{−5/4} has a minimum at x = (25/16)/a. Therefore ρ was assumed to be constant below (U − E1) = (25/16)/a."
I think people have simply used that "end-point" value (x = (25/16)/a) as the constant to be used. I think there is another approach i TALYS, see page 106 in the manual, that could possibly be interesting to look into as well.

**Comment by fzeiser** [comment]
> However, it is not clear what is happening with the spin cutoff parameter, though.

Yep, agree... By the way: Does eq. (4.257) look familiar ;P?

But back to the decisions. On
1. Yes, we could. I guess that's something like what Magne likes to do these days. ;P

2. This is there suggestion if the discrete scheme is incomplete, right. Here's the difference between the approximation of TALYS vs. the approximation of EB08 mentioned in EB09 (the red dashed lines in their plots) [which I mentioned in a previous post]. Here would be the difference between the two:
https://gist.github.com/fzeiser/70fbb5271019acc76dcb9011bd837ba3
![approx](https://user-images.githubusercontent.com/15156776/72661308-909aba80-39d8-11ea-9c3d-2ad4c5942a89.png)

3. For Gg, in the long run, we could implement taking the discrete levels instead of the level density formula. This would probably reduce the impact of some of the above choices.

**Comment by fzeiser** [comment]
Note: The equation numbers here are for talys manual v. 1.9.

**Comment by cecgresonant** [comment]
> Yep, agree... By the way: Does eq. (4.257) look familiar ;P?

Indeed!! :D  I just asked Magne, and Alexander Voinov and M. got some inspiration from this expression for the 'Alex' option in counting.c :) 


> 2. This is there suggestion if the discrete scheme is incomplete, right. Here's the difference between the approximation of TALYS vs. the approximation of EB08 mentioned in EB09 (the red dashed lines in their plots) [which I mentioned in a previous post]. Here would be the difference between the two:
>    https://gist.github.com/fzeiser/70fbb5271019acc76dcb9011bd837ba3
>    ![approx](https://user-images.githubusercontent.com/15156776/72661308-909aba80-39d8-11ea-9c3d-2ad4c5942a89.png)

This is very interesting! My first thought is that the TALYS curve is much more reasonable. For nuclei around A=50, there are experimental data indicating a spin cutoff of about 2.7, in much better agreement with the TALYS curve than the von Egidy & Bucurescu 2005/2006. 
 
> 3. For Gg, in the long run, we could implement taking the discrete levels instead of the level density formula. This would probably reduce the impact of some of the above choices.

I agree. This will also help us to get an excellent match with our input <Gg> and the TALYS-calculated <Gg> when we use this as a consistency check for e.g. (n, gamma) calculations.
